lib/components/popups/course-search/postreqs/PostReqSection.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useState, useEffect, FC } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  selectSemester,
+  selectYear,
+  updateSearchStack,
+  selectVersion,
+  selectInspectedCourse,
+} from '../../../../slices/searchSlice';
+import { getCourse, getCourseYear } from '../../../../resources/assets';
+import {
+  selectCurrentPlanCourses,
+  selectPlan,
+} from '../../../../slices/currentPlanSlice';
+import { selectCourseCache } from '../../../../slices/userSlice';
+import { selectCourseToShow } from '../../../../slices/popupSlice';
+import {
+  SISRetrievedCourse,
+  UserCourse,
+  Year,
+} from '../../../../resources/commonTypes';
+
+const PostReqSection: FC = () => {
+  const dispatch = useDispatch();
+  const version = useSelector(selectVersion);
+  const inspected = useSelector(selectInspectedCourse);
+  const currPlanCourses = useSelector(selectCurrentPlanCourses);
+  const semester = useSelector(selectSemester);
+  const year = useSelector(selectYear);
+  const currentPlan = useSelector(selectPlan);
+  const courseCache = useSelector(selectCourseCache);
+  const courseToShow = useSelector(selectCourseToShow);
+
+  const convertTermToInt = (term: string): number => {
+    if (term === 'fall') {
+      return 0;
+    } else if (term === 'intersession') {
+      return 2;
+    } else if (term === 'spring') {
+      return 3;
+    } else if (term === 'summer') {
+      return 4;
+    }
+    return 5;
+  };
+
+  /**
+   * Parses through the currentPlan years and returns year object corresponding to the year of the prereq or the current year if not found
+   * @param toShow - course to find year of
+   *  @returns year object
+   */
+  const getYearById = (toShow: UserCourse | null): Year => {
+    const yearToGet: string = toShow ? toShow.year_id : year;
+    for (const yearObj of currentPlan.years) {
+      if (yearObj._id === yearToGet) {
+        return yearObj;
+      }
+    }
+    return currentPlan.years[currentPlan.years.length - 1];
```

Comment:
TBH not sure how the logic works 100% since I took it from the pre-req function. However, I can confirm that it always returns the year corresponding to the popup.
When courseToShow is not null(this is when we click on a course already in the plan), it returns the year corresponding to that course.
When courseToShow is null(we just search for a course that isn't in the plan), it returns the year that we are searching for courses in and that line accomplishes that.

Reply:
ok... so maybe we rarely hit this line? not sure

Reply:
ill just trust you